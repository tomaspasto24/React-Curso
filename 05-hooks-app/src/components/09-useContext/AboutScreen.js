import React, { useContext } from 'react'
import { UserContext } from './UserContext'

export const AboutScreen = () => {

    const { setUser } = useContext(UserContext);

    const handleClick = () => {
        setUser( {} );
    };

    return (
        <div>
            <h1>AboutScreen</h1>
            <hr />

            <button
                className='btn btn-primary'
                onClick={ handleClick }
            >
                Logout
            </button>
        </div>
    )
}
