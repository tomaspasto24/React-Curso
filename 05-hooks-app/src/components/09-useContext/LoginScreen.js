import React, { useContext } from 'react';
import { UserContext } from './UserContext';


export const LoginScreen = () => {

    const { user, setUser } = useContext(UserContext);

    return (
        <div>
            <h1>LoginScreen</h1>
            <hr />
            
            <button
                className='btn btn-primary'
                onClick={ () => setUser({ id: 2404, name: 'Fernando'}) }
            >
                Login
            </button>
        </div>
    )
}
