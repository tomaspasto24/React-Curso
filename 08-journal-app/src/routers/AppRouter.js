import React, { useEffect } from 'react';
import {
    BrowserRouter as Router,
    Switch,
    Route,
    Redirect,
} from "react-router-dom";
import { JournalScreen } from '../components/journal/JournalScreen';
import { AuthRouter } from './AuthRouter';
import { firebase } from '../firebase/firebase-config';
import { useDispatch } from 'react-redux';
import { login } from '../actions/auth';

export const AppRouter = () => {

    const dispatch = useDispatch();
    
    // Es un efecto con una dependencia vacÃ­a por lo que solo se va a ejecutar una vez.(el dispatch es porque asi no queda como dependencia faltante).
    // Lo que hace es cargar el estado anterior, en caso de recargar el navegador. onAuthStateChanged siempre se va a estar ejecutando.
    useEffect( () => {
        firebase.auth().onAuthStateChanged( (user) => {

            if ( user?.uid ) {
                dispatch( login(user.uid, user.displayName) );
            }

        });
    }, [ dispatch ] );

    return (
        <Router>
            <div>
                <Switch>

                    <Route 
                        path='/auth' 
                        component={ AuthRouter } 
                    />
                    
                    <Route 
                        exact 
                        path='/' 
                        component={ JournalScreen } 
                    />

                    <Redirect
                        to='/auth/login'
                    />
                                    
                </Switch>
            </div>
        </Router>
    );
};
