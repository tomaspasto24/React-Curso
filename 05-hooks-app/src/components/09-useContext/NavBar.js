import React from 'react';
import { NavLink } from 'react-router-dom';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const NavBar = () => {
    return (
        <>
            <nav className="navbar navbar-expand-sm navbar-dark bg-dark">
                <div className="container-fluid">
                    <NavLink to='./' className="navbar-brand">useContext</NavLink>
                    <div className="collapse navbar-collapse" id="navbarNavAltMarkup">
                    <div className="navbar-nav">
                        {/* activeClassName establece una clase si el URL coincide con el path. */}
                        <NavLink to='./' className="nav-link" aria-current="page">Home</NavLink>
                        <NavLink to='./About' className="nav-link">About</NavLink>
                        <NavLink to='./Login' className="nav-link">Login</NavLink>
                    </div>
                    </div>
                </div>
            </nav> 
        </>
    )
}
