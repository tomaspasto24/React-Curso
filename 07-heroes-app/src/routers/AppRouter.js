import React from 'react';
import { Routes, Route, BrowserRouter } from 'react-router-dom';
import { LoginScreen } from '../components/login/LoginScreen';
import { DashboardRoutes } from './DashboardRoutes';
import { PrivateRoute } from './PrivateRoute';
import { PublicRoute } from './PublicRoute';

export const AppRouter = () => {
  return (
      <BrowserRouter>   
        <Routes>

            <Route path="/login" element={
              <PublicRoute>
                  <LoginScreen />
              </PublicRoute>
            } />
            
            {/* PrivateRoute se encarga de validar los datos de la variable user 
            para comrpobar si puede ver las rutas dentro de DashboardRoutes.
            Cuando se ingresa un componente dentro de otro, este termina ingresado en las props del componente padre.
            */}
            <Route path="/*" element={
              <PrivateRoute>
                 <DashboardRoutes />
              </PrivateRoute>
            } />

        </Routes>
      </BrowserRouter>
  );
};
