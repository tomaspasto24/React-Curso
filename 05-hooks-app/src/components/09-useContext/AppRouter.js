import React from 'react';
import {
    BrowserRouter as Router,
    Route,
    Routes,
} from 'react-router-dom';
import { NavBar } from './NavBar';
import { AboutScreen } from './AboutScreen';
import { HomeScreen } from './HomeScreen';
import { LoginScreen } from './LoginScreen';

export const AppRouter = () => {
    return (
        <Router>
          <div>

              <NavBar />
            
              <div className='container'>
                <Routes>

                        {/* exact es para que sea la direcci√≥n exacta. 
                        element se ingresa el componente que tiene que ser renderizado. */}

                        <Route exact path='/' element={ <HomeScreen /> } />
                        <Route exact path='/about' element={ <AboutScreen /> } />
                        <Route exact path='/login' element={ <LoginScreen /> } />

                        {/* el path='*', se usa como por defecto. */}
                        <Route path='*' element={ <HomeScreen /> } />

                </Routes>
              </div>
          
          </div>  
        </Router>
    )
}
