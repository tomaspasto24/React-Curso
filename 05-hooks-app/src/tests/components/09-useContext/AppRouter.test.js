import React from 'react';
import { mount } from 'enzyme';
import { AppRouter } from '../../../components/09-useContext/AppRouter';
import { UserContext } from '../../../components/09-useContext/UserContext';

describe('Pruebas sobre <AppRouter />.', () => {
    const user = {
        name: 'Tom√°s',
        email: 'tomaspasto24@gmail.com'
    };
    
    const wrapper = mount( 
    <UserContext.Provider value={ { user } }>
        <AppRouter /> 
    </UserContext.Provider>
    );

    test('Debe de mostrarse correctamente.', () => {
        expect(wrapper).toMatchSnapshot();
    });

});