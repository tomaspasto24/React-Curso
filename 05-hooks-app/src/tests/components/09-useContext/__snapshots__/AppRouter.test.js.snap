// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <AppRouter />. Debe de mostrarse correctamente. 1`] = `
<div>
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <div
      className="container-fluid"
    >
      <a
        aria-current="page"
        className="navbar-brand active"
        href="/"
        onClick={[Function]}
      >
        useContext
      </a>
      <div
        className="collapse navbar-collapse"
        id="navbarNavAltMarkup"
      >
        <div
          className="navbar-nav"
        >
          <a
            aria-current="page"
            className="nav-link active"
            href="/"
            onClick={[Function]}
          >
            Home
          </a>
          <a
            className="nav-link"
            href="/About"
            onClick={[Function]}
          >
            About
          </a>
          <a
            className="nav-link"
            href="/Login"
            onClick={[Function]}
          >
            Login
          </a>
        </div>
      </div>
    </div>
  </nav>
  <div
    className="container"
  >
    <div>
      <h1>
        HomeScreen
      </h1>
      <hr />
      <pre>
        {
   "name": "Tom√°s",
   "email": "tomaspasto24@gmail.com"
}
      </pre>
    </div>
  </div>
</div>
`;
